packages/app-server-issuer/src/exchange-provider/ExchangeProviderManager.ts

```diff
@@ -0,0 +1,136 @@
+import { number, bytes } from '@ckb-lumos/codec';
+import { Cell, Indexer, helpers, HexString, BI as BigNumber } from '@ckb-lumos/lumos';
+import { AcpTransferSudtBuilder } from '@ckitjs/ckit';
+import { ServerContext } from '../types';
+import { Config } from './config';
+
+export class ExchangeProviderManager {
+  private cells: Cell[] = [];
+  private initiated: boolean;
+  private config!: Config;
+  private context!: ServerContext;
+
+  private static sigleton: ExchangeProviderManager | undefined;
+
+  static getInstance(): ExchangeProviderManager {
+    if (ExchangeProviderManager.sigleton == undefined) {
+      ExchangeProviderManager.sigleton = new ExchangeProviderManager();
+    }
+
+    return ExchangeProviderManager.sigleton;
+  }
+
+  private constructor() {
+    this.initiated = false;
+  }
+
+  assertInitiated(): void {
+    if (!this.initiated) {
+      throw new Error('ExchangeProviderManager is not initiated');
+    }
+  }
+
+  private async refreshCells(): Promise<void> {
+    this.assertInitiated();
+
+    const indexer = new Indexer(this.context.ckitProvider.mercuryUrl, this.context.ckitProvider.rpcUrl);
+    const cells = (
+      await indexer.getCells({
+        script_type: 'lock',
+        script: this.context.ckitProvider.parseToScript(this.context.exchangeSigner!.getAddress()),
+        filter: {
+          script: this.context.ckitProvider.newScript('SUDT', this.config.sudtArgs),
+        },
+      })
+    ).objects;
```

Comment:
maybe it is unnecessary here, the [collectCells](https://github.com/nervosnetwork/ckit/blob/develop/packages/ckit/src/providers/mercury/MercuryProvider.ts#L155) is more friendly

```ts
await provider.collectCells(script_type: 'lock',
	        script: this.context.ckitProvider.parseToScript(this.context.exchangeSigner!.getAddress()),
	        filter: {
	          script: this.context.ckitProvider.newScript('SUDT', this.config.sudtArgs),
	        },
	      })
	    ))

```